x-pack/plugins/global_search/server/services/utils.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import type { IBasePath, KibanaRequest } from 'src/core/server';
+import type { IBasePath as BasePathAccessor } from '../../common/utils';
+
+export const getRequestBasePath = (
+  request: KibanaRequest,
+  basePath: IBasePath
+): BasePathAccessor => {
+  const requestBasePath = basePath.get(request);
+  return {
+    prepend: (path) => {
```

Comment:
is it possible to reuse `basePath.prepend` here?

Reply:
AFAIK there is no way to only retrieve the request-specific (`requestScopePath`) part of the basePath via the service, as `get` returns the 'whole' base path:

https://github.com/elastic/kibana/blob/bf04235dae35452061cc7ea3d86d96c19a58206c/src/core/server/http/base_path_service.ts#L43-L49

and `prepend` only allow to preprend using the server's base path, not a request-scoped basePath:

https://github.com/elastic/kibana/blob/bf04235dae35452061cc7ea3d86d96c19a58206c/src/core/server/http/base_path_service.ts#L70-L77

But maybe adding a new core API is better. I was surprised to be the first to have such need on the server-side TBH.

I can add something like `prependRequestPath(path: string, request: KibanaRequest)` to the server-side `BasePath`?